code/game/objects/machinery/miner.dm

```diff
@@ -0,0 +1,167 @@
+			if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+				return FALSE
+		playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
+		user.visible_message("<span class='notice'>[user] starts repairing [src]'s tubing and plating.</span>",
+		"<span class='notice'>You start repairing [src]'s tubing and plating.</span>")
+		if(do_after(user, 150, TRUE, src, BUSY_ICON_BUILD) && miner_status == MINER_SMALL_DAMAGE)
+			playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
+			miner_status = MINER_RUNNING
+			user.visible_message("<span class='notice'>[user] repairs [src]'s tubing and plating.</span>",
+			"<span class='notice'>You repair [src]'s tubing and plating.</span>")
+			update_icon()
+			return TRUE
+
+/obj/machinery/miner/examine(mob/user)
+	..()
+	if(ishuman(user))
+		switch(miner_status)
+			if(MINER_DESTROYED)
+				to_chat(user, "<span class='info'>It's heavily damaged, and you can see internal workings.</span>")
+				to_chat(user, "<span class='info'>Use a blowtorch, then wirecutters, then wrench to repair it.</span>")
+			if(MINER_MEDIUM_DAMAGE)
+				to_chat(user, "<span class='info'>It's damaged, and there are broken wires hanging out.</span>")
+				to_chat(user, "<span class='info'>Use a wirecutters, then wrench to repair it.</span>")
+			if(MINER_SMALL_DAMAGE)
+				to_chat(user, "<span class='info'>It's lightly damaged, and you can see some dents and loose piping.</span>")
+				to_chat(user, "<span class='info'>Use a wrench to repair it.</span>")
+			if(MINER_RUNNING)
+				to_chat(user, "<span class='info'>[src]'s internal storage currently has [stored_mineral] sheets stored.</span>")
+
+/obj/machinery/miner/attack_hand(mob/living/user)
+	var/mob/living/L = user
+	if(miner_status != MINER_RUNNING)
+		to_chat(L, "<span class='warning'>[src] is damaged!</span>")
+		return
+	if(stored_mineral != 0)
+		if(stored_mineral >= 50)	//We shouldnt get more, but just in case
+			stop_processing()
+		new mineral_produced(L.loc, stored_mineral)
+		stored_mineral = 0
+	else
+		to_chat(L, "<span class='warning'>[src]'s internal storage currently has no minerals stored!</span>")
+
+/obj/machinery/miner/process()
+	if(miner_status == MINER_RUNNING)
+		if(add_tick >= required_ticks && stored_mineral <= 49)	//make one phoron per 60 ticks, or 3 seconds.
+			stored_mineral += 1			
+			add_tick = 0
+		else if(stored_mineral >= 50)
+			stop_processing()
+		else
+			add_tick += 1
+
+/obj/machinery/miner/attack_alien(mob/living/carbon/xenomorph/M)
+	M.do_attack_animation(src, ATTACK_EFFECT_CLAW)
+	M.visible_message("<span class='danger'>[M] slashes \the [src]!</span>", \
+	"<span class='danger'>We slash \the [src]!</span>", null, 5)
+	playsound(loc, "alien_claw_metal", 25, 1)
+	switch(miner_status)
+		if(MINER_RUNNING)
+			stop_processing()
+			stored_mineral = 0
+			miner_status = MINER_SMALL_DAMAGE
+		if(MINER_SMALL_DAMAGE)
+			miner_status = MINER_MEDIUM_DAMAGE
+		if(MINER_MEDIUM_DAMAGE)
+			miner_status = MINER_DESTROYED

```

Comment:
Use a `set_miner_status(new_status)` proc so you can react to the changes and immediately stop/start processing as needed.